src/crypto/CryptoHash.cpp

```diff
@@ -29,27 +29,42 @@ class CryptoHashPrivate
 };
 
 CryptoHash::CryptoHash(CryptoHash::Algorithm algo)
+    : CryptoHash::CryptoHash(algo, false) {}
+
+CryptoHash::CryptoHash(CryptoHash::Algorithm algo, bool hmac)
     : d_ptr(new CryptoHashPrivate())
 {
     Q_D(CryptoHash);
 
     Q_ASSERT(Crypto::initalized());
 
-    int algoGcrypt;
+    int algoGcrypt = -1;
+    unsigned int flagsGcrypt = 0;
```

Comment:
@angelsl What about using the `GCRY_MD_FLAG_SECURE` flag by default? From the doc:
> Allocate all buffers and the resulting digest in "secure memory". Use this is the hashed data is highly confidential.

Reply:
Added that.